pkg/options/machinepool/create.go

```diff
@@ -0,0 +1,259 @@
+	)
+
+	flags.BoolVar(
+		&options.UseSpotInstances,
+		"use-spot-instances",
+		false,
+		"Use spot instances for the machine pool.",
+	)
+
+	flags.StringVar(
+		&options.SpotMaxPrice,
+		"spot-max-price",
+		"on-demand",
+		"Max price for spot instance. If empty use the on-demand price.",
+	)
+
+	flags.BoolVar(
+		&options.MultiAvailabilityZone,
+		"multi-availability-zone",
+		true,
+		"Create a multi-AZ machine pool for a multi-AZ cluster")
+
+	flags.StringVar(
+		&options.AvailabilityZone,
+		"availability-zone",
+		"",
+		"Select availability zone to create a single AZ machine pool for a multi-AZ cluster")
+
+	flags.StringVar(
+		&options.Subnet,
+		"subnet",
+		"",
+		"Select subnet to create a single AZ machine pool for BYOVPC cluster")
+
+	flags.StringVar(
+		&options.Version,
+		"version",
+		"",
+		"Version of OpenShift that will be used to install a machine pool for a hosted cluster,"+
+			" for example \"4.12.4\"",
+	)
+
+	flags.BoolVar(
+		&options.Autorepair,
+		"autorepair",
+		true,
+		"Select auto-repair behaviour for a machinepool in a hosted cluster.",
+	)
+
+	flags.StringVar(
+		&options.TuningConfigs,
+		"tuning-configs",
+		"",
+		"Name of the tuning configs to be applied to the machine pool. Format should be a comma-separated list. "+
+			"Tuning config must already exist. "+
+			"This list will overwrite any modifications made to node tuning configs on an ongoing basis.",
+	)
+
+	flags.StringVar(
+		&options.KubeletConfigs,
+		"kubelet-configs",
+		"",
+		"Name of the kubelet config to be applied to the machine pool. A single kubelet config is allowed. "+
+			"Kubelet config must already exist. "+
+			"This will overwrite any modifications made to node kubelet configs on an ongoing basis.",
+	)
+
+	flags.StringVar(&options.RootDiskSize,
+		"disk-size",
+		"",
+		"Root disk size with a suffix like GiB or TiB",
+	)
+
+	flags.StringSliceVar(&options.SecurityGroupIds,
+		securitygroups.MachinePoolSecurityGroupFlag,
+		nil,
+		"The additional Security Group IDs to be added to the machine pool. "+
+			"Format should be a comma-separated list.",
+	)
+
+	flags.StringVar(&options.NodeDrainGracePeriod,
+		"node-drain-grace-period",
+		"",
+		"You may set a grace period for how long Pod Disruption Budget-protected workloads will be "+
+			"respected when the NodePool is being replaced or upgraded.\nAfter this grace period, all remaining workloads "+
+			"will be forcibly evicted.\n"+
+			"Valid value is from 0 to 1 week (10080 minutes), and the supported units are 'minute|minutes' or "+
+			"'hour|hours'. 0 or empty value means that the NodePool can be drained without any time limitations.\n"+
+			"This flag is only supported for Hosted Control Planes.",
+	)
+
+	flags.StringSliceVar(
+		&options.Tags,
+		"tags",
+		nil,
+		"Apply user defined tags to all resources created by ROSA in AWS. "+
+			"Tags are comma separated, for example: 'key value, foo bar'",
+	)
+
```

Comment:
@den-rgb I think we're removing the `maxSurge` and `maxUnavailable` args from the CLI again (they were added back in #2259 Can you add them back please.

I'd like to get #2256 merged so we can be sure we're not dropping any other args as part of this refactor.

Reply:
done thanks for the reminder